### PR TITLE
Added SCDLPComplianceRule

### PR DESCRIPTION
**Pull request by NikCharlebois**
#### Pull Request (PR) description
Adds the new SCDLPComplianceRule resource

#### This Pull Request (PR) fixes the following issues
N/A

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/office365dsc/185)

### CODE REVIEW

**Comment by NikCharlebois** [comment]
Please hold off on review, will increase code coverage first

**Comment by NikCharlebois** [comment]
@ykuijs Ready for you review when you get a chance buddy

File: Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.psm1 [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Testing configuration of DLPComplianceRule for $Name"
+
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    Write-Verbose -Message "Current Values: $(Convert-O365DscHashtableToString -Hashtable $CurrentValues)"
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.psm1]: Is this line really necessary? The last line in the Get method also outputs this data, resulting in duplicate output.

File: Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.psm1 [hunk 2]
```diff
@@ -0,0 +1,559 @@
...
+    )
+
+    Write-Verbose -Message "Setting configuration of DLPComplianceRule for $Name"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+                      -Platform SecurityComplianceCenter
+
+    $CurrentRule = Get-TargetResource @PSBoundParameters
+
+    if (('Present' -eq $Ensure) -and ('Absent' -eq $CurrentRule.Ensure))
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.psm1]: Is there a specific reason that you placed the text on the left of the comparison? I know this is a best practice for a comparison with $null, but is this also true for other comparisons?

File: Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.schema.mof [hunk 3]
```diff
@@ -0,0 +1,36 @@
+[ClassVersion("1.0.0")]
+Class MSFT_SCDLPSensitiveInformation
+{
+    [Required, Description("Name of the Sensitive Information Content")] String name;
+    [Write, Description("Id of the Sensitive Information Content")] String id;
+    [Write, Description("")] String maxconfidence;
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.schema.mof]: No descriptions for these parameters?

File: Modules/Office365DSC/Modules/O365DSCReverse.psm1 [hunk 4]
```diff
@@ -826,6 +826,25 @@ function Start-O365ConfigurationExtract
     }
     #endregion
 
+    #region "SCDLPComplianceRUle"
```
Comment (ykuijs) [inline Modules/Office365DSC/Modules/O365DSCReverse.psm1]: (Picky mode) Typo: RUle needs to be Rule

**Comment by NikCharlebois** [comment]
:shipit:

**Pull request merged**